lib/configs/filesystem.ts

```diff
@@ -4,18 +4,37 @@
  */
 
 import type { Linter } from 'eslint'
+import type { ConfigOptions } from '../types.d.ts'
 
 /**
  * General config to exclude known non-source directories from linting
+ *
+ * @param options - configuration options
  */
-export const filesystem: Linter.Config[] = [
-	{
-		name: 'nextcloud/filesystem/ignores',
-		ignores: [
-			'**/dist',
-			'**/vendor',
-			'**/vendor-bin',
-			'**/package-lock.json',
-		],
-	},
-]
+export function filesystem(options: ConfigOptions): Linter.Config[] {
+	const config = [
+		{
+			name: 'nextcloud/filesystem/ignores',
+			ignores: [
+				'**/dist',
+				'**/vendor',
+				'**/vendor-bin',
+				'**/package-lock.json',
+			],
+		},
+	]
+
+	// apps have the l10n directory containing .js files not comply with the rules
+	// also the js directory contains compiled resources not comply with the rules
+	if (!options.isLibrary) {
+		config.push({
+			name: 'nextcloud/filesystem/app-ignores',
+			ignores: [
+				'js',
+				'l10n',
+			],
+		})
+	}
```

Comment:
Do we have any problems if we have a single config here for both apps and libraries?

Reply:
Normally not, currently it would not break any library.
Shall I just unify it?

Reply:
I'd keep it simple, yes. Without adding a config factory with a new group, where as simple static list just works